homeassistant/components/rointe/__init__.py

```diff
@@ -0,0 +1,74 @@
+"""The Rointe Heaters integration."""
+from __future__ import annotations
+
+from rointesdk.rointe_api import ApiResponse, RointeAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    CONF_INSTALLATION,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DOMAIN,
+    LOGGER,
+    PLATFORMS,
+    ROINTE_COORDINATOR,
+)
+from .coordinator import RointeDataUpdateCoordinator
+from .device_manager import RointeDeviceManager
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Rointe Heaters from a config entry."""
+
+    coordinator = await init_device_manager(hass, entry)
+
+    # Initialize Hass data if necessary.
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {}
+
+    hass.data[DOMAIN][entry.entry_id][ROINTE_COORDINATOR] = coordinator
+
+    await coordinator.async_config_entry_first_refresh()
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry and removes event handlers."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def init_device_manager(
```

Comment:
Totally up to you, but since this method is now pretty tight/focused, does it make sense to keep it as a separate method (vs. just putting the contents in `async_setup_entry`)?

No pressure there; just thinking.